armi/operators/tests/test_inspectors.py

```diff
@@ -52,6 +57,50 @@ def defdef(x, y, z):
 
         self.assertEqual(str(query), "<Query: beepbopboopbeep>")
 
+    def test_overwriteSettingsCorrectiveQuery(self):
+        """
+        Tests the case where a corrective query is resolved.
+        Checks to make sure the settings file is overwritten with the resolved setting.
+        """
+        # load settings from test settings file
+        self.cs["cycleLength"] = 300.0
+        self.cs.writeToYamlFile(self.filepathYaml)
+        self.cs.loadFromInputFile(self.filepathYaml)
+        self.assertEqual(self.cs["cycleLength"], 300.0)
+
+        # define corrective query
+        def csChange(x, y, z):
+            x[y] = z
+
+        self.inspector.addQuery(
+            lambda: self.inspector.cs["cycleLength"] == 300.0,
+            "Changing `cycleLength` from 300.0 to 666",
+            ":D",
+            lambda: csChange(self.cs, "cycleLength", 666),
+        )
+
+        # redefine prompt function in order to circumvent need for user input
+        def fakePrompt(*inputs):
+            return True
+
+        nominalPromptFunction = settingsValidation.prompt
+        settingsValidation.prompt = fakePrompt
+
+        # run inspector
+        self.inspector.run()
+
+        # check to see if file was overwritten correctly
+        self.cs.loadFromInputFile(self.filepathYaml)
+
+        self.assertEqual(self.cs["cycleLength"], 666)
+
+        # reset prompt function to nominal
+        settingsValidation.prompt = nominalPromptFunction
+
+        # reset test settings file
+        self.cs["cycleLength"] = 300.0
+        self.cs.writeToYamlFile(self.filepathYaml)
```

Comment:
Okay, this block is super important.  I like that it exists.

I almost feel like we should wrap the rest of this test in a try/except/finally, and put this in the finally, to make sure it gets run, even if the test fails.

Thoughts?

Reply:
Now that I am looking at it, maybe we don't actually need this block. The test starts by generating the test settings file, so even if it wasn't overwritten, the test starts by writing the file anyways. What do you think?

Reply:
If the test runs 2 or 3 times in a row without this block, great, remove it. Sure.